### DISCUSSION TITLE
Welcome to plex_debrid Discussions!

### DISCUSSION THREAD

**Discussion by itsToggle** [itsToggle/plex_debrid#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.

**Comment by Bayharbourbutcher1** [comment]
Sorry not sure if its the right place to comment (new to github) 

It would be great to have an option to restrict content size e.g less than 50gb for a movie, for scraped media in the script. Also possibly set filters like exclude dolby vision content etc.

**Comment by Yao-ming23** [comment]
Hey, i have gotten to the last step to actually run the script but im getting an error 

C:\Users\Ming\Desktop\Rclone prebuilt>python3 plex_rd.py
  File "C:\Users\Ming\Desktop\Rclone prebuilt\plex_rd.py", line 61
    <title>plex_debrid/plex_rd.py at main · itsToggle/plex_debrid · GitHub</title>
                                          ^
SyntaxError: invalid character '·' (U+00B7)

**Reply by itsToggle** [reply]
You need to change the name of that folder you created, it has an invalid character in its name :)

**Reply by Yao-ming23** [reply]
thanks for the reply! 

I didnt create any folder with that character though. I created a folder called 'Rclone prebuilt' with the files inside there. Or am i doing something wrong? sorry if im misunderstanding something.

If i edit the plex_rd.py file with notepad ++ and remove that invalid character, it keeps giving me a syntax error but just different places in the file

**Reply by Yao-ming23** [reply]
If anyone can offer help i would really appreciate it. I'm kind of stumped right now but would love to get this working

**Reply by itsToggle** [reply]
Im sorry, I totally misread the error! 
It looks like the plex_rd file has not been downloaded corrently, as it contains this "<title> ..." Line. It almost looks like you downloaded the HTML Page instead of cloning this repository :)

**Reply by Yao-ming23** [reply]
Sorry for the late reply.

You're totally right, my noob self didn't clone the repo correctly. 

So I was able to get it working, for the most part, I'm sorry I'm coming at you with all these issues! 

when I start up the script I'm first greeted with the following error:

Type 'exit' to return to the main menu.
[2022-07-10 13:18:47.258930] getting plex library section/s "4,3,2" ... [2022-07-10 13:18:47.289186] plex error: couldnt reach local plex library section '2' at server address: http://localhost:32400 - or this library really is empty.
done

So the way I set up everything was "movies" and "shows" on the rclone and a local folder on my pc. Added the sections in the settings files and shows up on my plex, then after seeing the error again I tried copying a small video file over to the rclone (not sure if that would even do anything) but still got the error, BUT I am still able to add movies to the watchlist and stream them. 

So thanks for this, it's really what I have been wanting for a while, I cant wait to see how amazing you make the rest of this release

**Reply by itsToggle** [reply]
Nice! That error is just telling you that the library section "2" does not have any content in it (or that it couldnt be reached by plex_debrid). I dont really have a way of destinguishing between a truly empty library and plex not being reachable, which is why this warning appears.

Which library does this section number ('2') correspond to? Is it the local folder on your pc or one of libraries of your rclone drive? If that library section really is empty, the warning will disappear once you add some content to it. But if there should be content in section '2', we need to do some investigating.

Depending on your debrid service, write-capabilities might be limited (so copying might not work). RealDebrid for example has no write capabilities at all, so copying something to the drive will not work.

Edit: Also, for future errors, please use the "Issues" page ;)

**Reply by Yao-ming23** [reply]
My apologies, just thought i wouldnt add an "issue" as small as that, but if you would like me to i will add that issue there and we can discuss it further?

**Comment by loki131** [comment]
Just stumbled on this project and wow! I do have a couple of questions...

Is there a way to install your rclone fork on a Synology NAS?
Can Trakt integration be used so managed users can add content?

I'm sure I'll have other questions after I install! :)

**Reply by itsToggle** [reply]
The rclone fork is available for Linux and FreeBSD, so im sure it'll work on a Synology NAS aswell :) There is also the option of utilizing the official rclone software, which will definitly work on your NAS. There are some limitations regarding realdebrid though - torrents with special characters are not listed, its generally slower than my fork, you cant delete stuff, etc.

There is a Trakt Integration (See readme.md), but to be honest Im not too proud of it. Its pretty hard to accuratly match content from Trakt to Plex, since Plex Discover does not allow for a search by IMDB ID. I did manage to get it working through a lot of hoops, but at the moment the Trakt watchlist is only updated every 30 minutes. I can probably increass the poll frequency though.

I just told my home users to create a free Plex account and invited them to my server so they can use all the Plex Pass features. That way they can enjoy the searching and instantanious streaming all inside Plex without the need of third party apps like Trakt :)

**Reply by itsToggle** [reply]
updated the trakt integration, plex_debrid now checks for new trakt content from users watchlists or public lists every 5 seconds.

**Comment by dorinvlas** [comment]
any idea why? (will delete this post in case I find a solution)
Happening on a ubuntu LXC(linux container) that I have on my proxomox node... plex media server and rclone is 100% functional, its just this error below. Thanks!

`[2022-07-18 12:31:17.599315] getting entire plex library ... done
[2022-07-18 12:31:17.608584] getting all plex watchlists ... done
[2022-07-18 12:31:21.721734] getting all overseerr requests ... Exception in thread Thread-1 (run):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "/root/plex_rd.py", line 2200, in run
    overseerr_requests = overseerr.requests()
  File "/root/plex_rd.py", line 1080, in __init__
    for element in response.results:
AttributeError: 'NoneType' object has no attribute 'results'`

**Reply by dorinvlas** [reply]
@itsToggle BTW this integration is awesome! its first time for me to use plex... and I am starting to use right away with debrid and your intergration on top... so I think its very promising for all movie fans.

**Reply by itsToggle** [reply]
Hey, looks like i didnt catch an exception there, which I will fix asap. The error is related to overseerr. Are you trying to integrate overseerr?

**Reply by dorinvlas** [reply]
hey. not sure about the overseerr... I am using this whole setup as prescribed in README here... with some adjustments so it works on LXC container... I did not enabled overseerr or configured it manually.

Anything that I should check? and where?

Thanks for such quick reply!

**Reply by itsToggle** [reply]
ah never mind, totally my mistake. The script tried to check for overseerr requests without you setting up the optional overseerr integration - so nothing wrong on your end. Ive updated the script now with a fix :)

**Reply by dorinvlas** [reply]
ha :) let me try the new one then... thanks

**Reply by dorinvlas** [reply]
did some quick testing and it looks good. Will test later with my kids for some nice colored fairy tale from big guys ;)
Thanks @itsToggle for all the social involvement ;)

**Reply by itsToggle** [reply]
awesome! enjoy :)

**Reply by dorinvlas** [reply]
BTW... not really sure how to use jackett in tandem plex_debrid an some private tracker(on the list of supported)? is there any hint or maybe you can point me to some sort of howto where I can figure out how I can use such solution in my current plex_debrid setup? 

Thanks.

**Reply by itsToggle** [reply]
ive written a little bit about this in the readme ('How to integrate your favorite indexers'), but you basically:

- install jackett
- open jacketts webui
- setup your private tracker
- open plex_debrid
- go to "/Settings/Scraper/Sources/Edit/Add" , choose 'jackett'
- when asked, enter the jackett base url (usually you can use the default here)
- when asked, enter the jackett api key (top right in jacketts webui)
- leave all other options empty
- done

Be aware though, that some private indexers do not allow the usage of debrid services that dont seed torrents (basically any debrid providers except debridlink). Using non-seeding debrid services on private indexers will negatively affect your ratio. If want to use a specific debrid service for a specific tracker, you can define this by navigating to /Settings/Debrid Services/Tracker specific debrid service'

**Comment by emigabi** [comment]
Hi. If you setup the debrid library through a device is it available on other devices, or you need to setup the scrip on every device? Specifically: setup the script on android tablet and use the library on LG tv with webOS.

**Reply by itsToggle** [reply]
Hi, im not sure what you mean - you only need to set up your Plex server, rclone and the plex_debrid script on one device. This then allows you to stream from your Plex server and download new content from within any Plex client on any device.

**Reply by emigabi** [reply]
Thank you. I think I get it...setup a laptop as a server (in my case) and then install the script on the laptop/server and use it as a bridge to Real-Debrid for my TV. Does the server need to transcode the video file or can it just pass it straight to my TV?

**Reply by itsToggle** [reply]
Yeah that's it. You can run the server on your Laptop, a NAS, an Nvidia shield, or even on a cloud computer (VPS) - join our discord server for suggestions, a lot of people are using a completely free to use cloud computer :) . 
If the server needs to transcode depends entirely on the Plex client you are using. Most clients can play the typical formats you will download directly, so the stream is so called "direct play". I would assume that direct play will work 99% of the time for your TV. If you watch through your browser on your computer, there is a good chance Plex will need to transcode though. You can read more about this on official Plex sources :)

**Reply by emigabi** [reply]
Thank you again. Never tried VPS, but I have thew things in the house (laptop, Nvidia shield) and see how it goes. I'll return if I have anymore questions. Again thanks.

**Comment by choman198** [comment]
Hello there,
First of all, incredible setup and allow me to thank you for all your work to make this intergration work. I have a fully functioning Plex server that took me less than 30 mins to setup with access to unlimited movies and shows and that is all thanks to you. I wanted to ask what the expected speed is from the Rclone fork. I seem to have some performance issues in general, movies taking 8-10 seconds to start loading and I know it is due to the Rclone mount since even browsing to the folders seems to be really slow sometimes. 

I am using a Windows server 2022 but I have also tried with Windows 10 and Ubuntu as well. I noticed ubuntu speeds were slightly better but I had issues with Rclone crashing or randomly disconnecting which is why i switched back to Windows. Again everything works great except for the really slow Real Debrid Mounted disk. Any way to improve this or am I doing something wrong?
